be/log.c

```diff
@@ -916,7 +921,8 @@ static void be_log_header_update(struct m0_be_log *log)
 		index = log->lg_unplaced_pos;
 		size  = log->lg_unplaced_size;
 	}
-	m0_be_log_header__set(&log->lg_header, log->lg_discarded, index, size);
+	m0_be_log_header__set(&log->lg_header, log->lg_discarded, index,
+			      size, log->lg_header.flh_serial_num);
```

Comment:
Are you sure, the "log->lg_header.flh_serial_num" is valid at this moment?

This is confusing. "log->lg_header" is updated with the serial number from itself.